Sources/JS/initTypeCollection.js

```diff
@@ -12,89 +12,71 @@
-        }
-    }
-    if(this.outputType != null)  result.outputType = this.outputType.serialize()
+    if (this.group === otherType.group) newType.group = this.group
+    newType.properties = arrayIntersection(this.properties, otherType.properties)
+    newType.methods = arrayIntersection(this.methods, otherType.methods)
 
-    return result
+    return newType
 }
 
 // variableNumber: Type
 var types = {}
-// Must be kept in sync with TypeSystem.swift
-baseTypes = {
-    nothing: 0,
-    undefined: 1 << 0,
-    integer: 1 << 1,
-    float: 1 << 2,
-    string: 1 << 3,
-    boolean: 1 << 4,
-    object: 1 << 5,
-    function: 1 << 6,
-    constructor: 1 << 7,
-    unknown: 1 << 8,
-    bigint: 1 << 9,
-    regexp: 1 << 10,
-}
-function updateType(number, value){
-    function setType(rawValue) {
-        types[number] = new Type()
-        types[number].definiteType = rawValue
-        types[number].possibleType = rawValue
+function getBaseType(rawValue) {
+    var newType = new Type()
+    newType.definiteType = rawValue
+    newType.possibleType = rawValue
+    if (rawValue === baseTypes.object) {
+        newType.properties = []
+        newType.methods = []
     }
+    return newType
+}
+function getCurrentType(value){
     try {
         // Fuzzilli handles both null/undefined as undefined
-        if (value == null) {
-            setType(baseTypes.undefined)
-            return
-        }
-        if (Number.isInteger(value)) {
-            setType(baseTypes.integer)
-            return
-        }
-        if (typeof value === 'number') {
-            setType(baseTypes.float)
-            return
-        }
-        if (typeof value === 'string') {
-            setType(baseTypes.string)
-            return
-        }
-        if (typeof value === 'boolean') {
-            setType(baseTypes.boolean)
-            return
-        }
-        if (typeof value === 'bigint') {
-            setType(baseTypes.bigint)
-            return
-        }
+        if (value == null) return getBaseType(baseTypes.undefined)
+        if (Number.isInteger(value)) return getBaseType(baseTypes.integer)
+        if (typeof value === 'number') return getBaseType(baseTypes.float)
+        if (typeof value === 'string') return getBaseType(baseTypes.string)
+        if (typeof value === 'boolean') return getBaseType(baseTypes.boolean)
+        if (typeof value === 'bigint') return getBaseType(baseTypes.bigint)
         try {
-            if (value instanceof RegExp) {
-                setType(baseTypes.regexp)
-                return
-            }
+            if (value instanceof RegExp) return getBaseType(baseTypes.regexp)
         } catch(err) {}
-
+        if (typeof value === 'object' || typeof value === 'symbol') {
+            var currentType = getBaseType(baseTypes.object)
+            setGroup(value, currentType)
+            while (value != null) {
+                var propertyNames = Object.getOwnPropertyNames(value)
```

Comment:
Maybe also needs to be in try-catch, same as above

Reply:
Similar question as above, how do we handle this? return type object without properties? or return unknown and let abstractinterpreter decide?

Reply:
I guess in this situation runtime type does not give us much information, so wide catch for unknown is sufficient